mtp_api/apps/payment/management/commands/clear_abandoned_payments.py

```diff
@@ -0,0 +1,42 @@
+from datetime import timedelta
+import textwrap
+
+from django.core.management import BaseCommand, CommandError
+from django.utils.timezone import now
+
+from credit.models import Credit
+from payment.models import Payment
+
+
+class Command(BaseCommand):
+    """
+    Delete payments that are still pending after at least a day.
+    These are abandoned payments that never succeeded to pass through GOV.UK Pay.
+    """
+    help = textwrap.dedent(__doc__).strip()
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument('--age', type=int, default=7, help='The minimum age of the payment in days')
+
+    def handle(self, *args, **options):
+        verbosity = options.get('verbosity', 1)
+        age = options['age']
+        if age < 1:
+            raise CommandError('Payment age must be at least 1 day')
+        abandoned_payments = Payment.objects.abandoned(now() - timedelta(days=age))
+        abandoned_count = abandoned_payments.count()
```

Comment:
In this case I think it is better to do ```len(abandoned_payments)``` because you get them afterwards anyway, so this would be doing a count followed by a select, rather than just the select if you use ```len```.